Create tee attributes.md (#42)
confidential-computing/governance: tee attributes.md

## TEE Attributes

Required properties to be called a “TEE”:
* ___Data integrity___ - data values stored or being operated upon within the bounds of the TEE cannot be modified by an actor operating outside the TCB.
* ___Data confidentiality___ - data values stored or being operated upon within the bounds of the TEE cannot be determined by an actor operating outside the TCB.
* ___Code integrity___ - software codes executed inside the bounds of the TEE cannot be modified by an actor operating outside the TCB.

TEE's may have other properties:

> * _Code confidentiality_ - software codes executed inside the bounds of the TEE cannot be modified by an actor operating outside the TCB.

Reviewer: Looks like a cut and paste error from line 6.  The definition should be different.
```suggestion
* _Code confidentiality_ - software codes executed inside the bounds of the TEE cannot be determined by an actor operating outside the TCB.
```

> * _Programmability_ - the extent to which the software codes operated upon inside the bounds of the TEE can be provided by an identified actor.

Reviewer: "can be provided" I think should be explicitly "after time of manufacture" or something similar.

Author: ```suggestion
* _Programmability_ - the extent to which the software codes operated upon inside the bounds of the TEE provided after time of manufacturer by an identified actor.
```

> * _Programmability_ - the extent to which the software codes operated upon inside the bounds of the TEE provided after time of manufacturer by an identified actor.

Reviewer: The others above are phrased as a sentence.   This one is phrased as a definition.  I'd recommend picking one style and being consistent.

> * _Attestability_ - the mechanism for demonstrating the contents of a TEE is infact secured by a TEE.

Reviewer: ```suggestion
* _Attestability_ - the mechanism for demonstrating the contents of a TEE is in fact secured by a TEE.
```

Reviewer: I'd also compare the text currently in the whitepaper, which might have a better definition

> * _Attestability_ - the mechanism for demonstrating the contents of a TEE is in fact secured by a TEE.

Reviewer: Grammar: "contents ... is" doesn't seem to match

* _Unspoofability (un-clonable) / non-repudiability)_ - when using cryptographic authentication mechanisms in such a manner as it difficult to deny the origin of an assertion (often in referring to identity tied to a transaction).

> * _Authenticated launch_ - (aka secure boot) a mechanism for determining that an approved TCB via cryptographic signature is established. As opposed to “measured” launch which just establishes which elements compose the TCB without determination of its origin or acceptability.

Reviewer 2: Do we mean of code _within_ or _external to_ the TEE?

Author: inside <--> within

> * _Recoverability_ if an error in the TCB of the TEE is discovered the ability for the error to be mitigated. Full recoverability allows for the attestation of the recovered TCB.

Reviewer: ```suggestion
* _Recoverability_ - if an error in the TCB of the TEE is discovered the ability for the error to be mitigated. Full recoverability allows for the attestation of the recovered TCB.
```

> * _Isolation_ - A mechanism by which the main properties of data & code integrity can be achieved.

Reviewer: The others are properties, this seems to be phrased as a mechanism, not a property

> * _Robustness_ - Actors operating outside the TCB may be defined as a software actor (e.g. the host operating system) or a hardware actor (e.g. the platform operator). The extent to which a TEE provides protection from one actor or another is defined by its TCB.

Reviewer: I'm having a hard time understanding this one.
